### PR TITLE
Cleaning up port/protocol split/validation code

### PR DESCRIPTION
**Pull request by coreydaley**
Cleaning up and de-duplicating port/protocol splitting and validation code

### CODE REVIEW

**Comment by coreydaley** [comment]
@openshift/sig-developer-experience ptal

File: pkg/util/portutils/common_test.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		},
+
+		{
+			port: "88080/tcp",
+			err:  "port number must be in range 0 - 65535",
+		},
+		{
+			port: "8080/xyz",
+			err:  "protocol must be tcp or udp",
+		},
```
Comment (bparees) [inline pkg/util/portutils/common_test.go]: add a test for a port like "$myarg"

Comment (coreydaley) [reply pkg/util/portutils/common_test.go]: Updated

**Comment by bparees** [comment]
/approve
/hold

(post-3.7)

**Comment by coreydaley** [comment]
/retest

On Nov 14, 2017 5:39 PM, "OpenShift CI Robot" <notifications@github.com>
wrote:

@coreydaley <https://github.com/coreydaley>: The following test *failed*,
say /retest to rerun them all:
Test name Commit Details Rerun command
ci/openshift-jenkins/extended_conformance_install_update 37bd4c2
<https://github.com/openshift/origin/commit/37bd4c2e0f880eba0497203974c99152e15987f2>
link
<https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/17184/test_pull_request_origin_extended_conformance_install_update/9404/>
/test
extended_conformance_install_update

Full PR test history <https://openshift-gce-devel.appspot.com/pr/17184>. Your
PR dashboard <https://openshift-gce-devel.appspot.com/pr/coreydaley>.
Please help us cut down on flakes by linking to
<https://github.com/kubernetes/community/blob/master/contributors/devel/flaky-tests.md#filing-issues-for-flaky-tests>
an open issue
<https://github.com/openshift/origin/issues?q=is:issue+is:open> when you
hit one in your PR.

Instructions for interacting with me using PR comments are available here
<https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md>.
If you have questions or suggestions related to my behavior, please file an
issue against the kubernetes/test-infra
<https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
repository. I understand the commands that are listed here
<https://github.com/kubernetes/test-infra/blob/master/commands.md>.

—

You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/17184#issuecomment-344423376>,
or mute
the thread

**Pull request merged**